### PR TITLE
Alerting: Add Notification error feedback on contact points view

### LINKED ISSUE #300: Ability to shift time in timepicker
In zabbix timepicker has nice ability to shift time back and forth:

zabbix

On the top you choose time range, on the bottom you may shift timespan. This is quite useful to compare current day with previous or even current hour today with the same hour yesterday.

I think it would be nice to have the same thing in grafana as well.

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**PR Title:**
Error notification feedback on Contact Points page

**Why do we need this feature?**
Sometimes notifications fail to be sent or notifications are not received, and previously there was no easy way of knowing what the error was or why the notification failed to deliver.

**What is this feature?**
Makes it easy for users to see that something is wrong with their contact point(s) and what is wrong (the reason for the error).
It uses a new API - the Receivers API `/api/alertmanager/grafana/config/api/v1/receivers 
`. For more information about the new endpoint used: [PR](https://github.com/grafana/grafana/pull/55429)
Each receiver has information on the error, eg. a time stamp (when the notification attempt was made) and duration of the attempt, and the error.
On the UI, users can now view the errors for each contact point - and if a contact point has more than one integration - for each integration. A new Health column shows the status Error, No Attempts, or OK.
A message at the right-hand corner of the screen alerts the user to the fact that there are errors and how many.
**only works with Grafana Alertmanager for now. (because it's the only alert manager that provides this new endpoint so far)

**Who is it for?**

- Any user who works on creating notification policies and contact points
- Any user who creates alert rules and wants to see if contact points are failing

**Which issue(s) this PR fixes**:

Fixes [#300](https://github.com/grafana/alerting-squad/issues/300)

**Special notes for your reviewer**:
The last two commits are the ones that should be reviewed/approved. The rest are already approved. 

https://user-images.githubusercontent.com/33540275/193751624-5d47f079-e6f4-4411-ba53-1269595ba791.mp4

### CODE REVIEW

**Comment by VikaCep** [comment]
While trying it locally I noticed we're making calls to this new endpoint and showing the error message in the contact point creation form:

image

Is it supposed to be there?

**Comment by soniaAguilarPeiron** [comment]
> While trying it locally I noticed we're making calls to this new endpoint and showing the error message in the contact point creation form:
> 
> image
> 
> Is it supposed to be there?

Good point @VikaCep! I think we shouldn't, as this information makes no sense when creating/updating a contact point.

File: public/app/features/alerting/unified/api/grafana.ts [hunk 1]
```diff
@@ -1,6 +1,78 @@
...
+    (prevCount: number, receiverState: ReceiverState) => prevCount + receiverState.errorCount,
+    0
+  );
+  return { ...contactPointsState, errorCount: errorsCount };
+};
+
+export const getIntegrationType = (integrationName: string): string | undefined =>
+  parseIntegrationName(integrationName)?.type;
+
+export async function fetchContactPointsState(alertManagerSourceName: string): Promise<ContactPointsState> {
```
Comment (konrad147) [inline public/app/features/alerting/unified/api/grafana.ts:65]: nit: If you want, you can try to use RTK Query for fetching data

**Review by konrad147 — APPROVED** [review]
Looks great! Just some non-blocking nits

File: public/app/features/alerting/unified/Receivers.tsx [hunk 2]
```diff
@@ -17,14 +19,42 @@ import { ReceiversAndTemplatesView } from './components/receivers/ReceiversAndTe
...
+} from './state/actions';
+import { CONTACT_POINTS_STATE_INTERVAL_MS } from './utils/constants';
 import { GRAFANA_RULES_SOURCE_NAME } from './utils/datasource';
 import { initialAsyncRequestState } from './utils/redux';
 
+export interface NotificationErrorProps {
+  errorCount: number;
+}
+
+const NotificationError: FC<NotificationErrorProps> = ({ errorCount }: NotificationErrorProps) => {
```
[the remark this answers is not in the document]
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/Receivers.tsx:35]: true!!👀

**Review by gillesdemey — APPROVED** [review]

**Pull request merged**